test/e2e/tests/confirmations/contract-interaction-redesign.spec.js

```diff
@@ -0,0 +1,144 @@
+const { hexToNumber } = require('@metamask/utils');
+const {
+  defaultGanacheOptions,
+  defaultGanacheOptionsForType2Transactions,
+  openDapp,
+  unlockWallet,
+  WINDOW_TITLES,
+  withFixtures,
+} = require('../../helpers');
+const FixtureBuilder = require('../../fixture-builder');
+const { CHAIN_IDS } = require('../../../../shared/constants/network');
+
+describe('Confirmation Redesign Contract Interaction Component', function () {
+  if (!process.env.ENABLE_CONFIRMATION_REDESIGN) {
+    return;
+  }
+
+  it(`Opens a contract interaction type 0 transaction (Legacy)`, async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            preferences: { redesignedConfirmationsEnabled: true },
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+        await openDapp(driver);
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+        await driver.clickElement(`#deployButton`);
+
+        await driver.delay(1000);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        await driver.waitForSelector({
+          css: '.confirm-page-container-summary__action__name',
+          text: 'Contract deployment',
+        });
+
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+        await driver.clickElement({ text: 'Activity', tag: 'button' });
+        await driver.waitForSelector(
+          '.transaction-list__completed-transactions .activity-list-item:nth-of-type(1)',
+        );
```

Comment:
could we skip all these steps and just deploy the contract on with the ganache seeder and pass this into the fixtures?

This way we can go directly to the contract interaction step

`smartContract: SMART_CONTRACTS.PIGGYBANK,`